proposals/0000-reduce-fragmentation.md

```diff
@@ -190,151 +339,448 @@ const autoCompleteIOS = {
-  * See [animation](https://developer.mozilla.org/en-US/docs/Web/CSS/animation)).
-* CSS Transitions
-  * Support declarative transitions that can be optimized on the native side and avoid the need for `Animated`.
-  * Support `transitionProperty`, `transitionDuration`, `transitionTimingFunction`, and `transitionDelay` (see [transition](https://developer.mozilla.org/en-US/docs/Web/CSS/transition)).
-* `aspectRatio`
-  * Support string values, i.e., `'16 / 9'`, to align with CSS.
-* `backgroundImage`
-  * Support setting background images via `url()`.
-  * Optional: support CSS gradients.
-* `boxShadow`
-  * Add native support for CSS box shadows.
-  * This would replace the very buggy iOS-specific `shadow*` styles and Android `elevation` style, allowing developers to unify shadows across native and web platforms.
-* `filter`
- * Support declarative filters per [filter](https://developer.mozilla.org/en-US/docs/Web/CSS/filter)
-* `fontVariant`
-  * Support space-separated string values to align with CSS.
-* `fontWeight`
-  * Support number values to align with React DOM / CSS.
-* `lineClamp`
-  * Support CSS clamping text to a set number of lines via style.
-  * This would be equivalent to using the `numberOfLines` prop on `<Text>` components.
-* `objectFit`
-  * Support CSS `objectFit` property for `<Image>`.
-  * This would be a subset (no "repeat" keyword) of functionality currently supported by the `resizeMode` prop and style for `<Image>`.
-  * `objectFit === 'contain'` => `resizeMode = 'contain'`
-  * `objectFit === 'cover'` => `resizeMode = 'cover'`
-  * `objectFit === 'fill'` => `resizeMode = 'stretch'`
-  * `objectFit === 'scale-down'` => `resizeMode = 'contain'`
-* `pointerEvents`
-  * Support setting the `pointerEvents` values via style rather than a prop.
-  * Retain the React Native specific `box-none` and `box-only` values.
-* `textShadow`
-  * Add native support for CSS text shadows.
-* `transform`
-  * Support using string values to set transforms.
-  * This can be shimmed in user-space with a parser.
-* `verticalAlign`
-  * Make `verticalAlign` the same as `textAlignVertical`.
-  * `verticalAlign === 'middle'` is the same as `textAlignVertical = 'center'`;
-* `userSelect`
-  * Support setting ability to select text via style.
-  * Equivalent to using `selectable` prop on `<Text>`.
+-----
+
+# Styles API (CSS subset)
+
+## StyleSheet
+
+* [ ] Deprecate `StyleSheet.absoluteFill`. Use `position:'absolute';inset:0;` instead.
+* [ ] Deprecate `StyleSheet.absoluteFillObject`.
+* [ ] Deprecate `StyleSheet.compose()`. Use existing `[ a, b ]` syntax instead.
+* [ ] Deprecate `StyleSheet.flatten()`. This API encourages runtime introspection of styles in render calls. This pattern is a performance cost (flattening arrays of objects), and prevents us from supporting build-time optimizations for web (e.g., extracting styles to CSS files).
```

Comment:
Could be made internal-only too. Flattening is required as part of the native interface / React's prop diffing.

Reply:
Yeah, as long as it's not being used in user space

Reply:
When working on a library, it's useful to be able to accept a regular-style prop and flatten it to pass it down to the underlying native component. If this is removed, then we'd need to either flatten manually or force the user to pass a plain object for `style` - which isn't ideal since the different kind of API won't be intuitive, or user would need to manually spread the objects.

Reply:
On web we don't need the individual style properties as passing the class name to the tag takes care of applying multiple styles.

With a native component (e.g. if I implement my own `Image` component), we want a regular object that we use to apply the styles on the native side. It's the same reason why React Native core does it for the built-in components.

Reply:
That's something that could be provided without having JS `flatten` exist in user space and being used for other purposes that cause performance issues.

Reply:
Yeah, just something that needs to exist before removing this API

Reply:
Deprecation != removal. And flattening is very straight-forward, so you could definitely do this / copy-paste in user-space too.